source/Octopus.Tentacle.Tests.Integration/Support/RunningTentacle.cs

```diff
@@ -54,11 +59,41 @@ public async Task Stop(CancellationToken cancellationToken)
             await t;
         }
 
+        private void StopOnDispose(CancellationToken cancellationToken)
+        {
+            if (cancellationTokenSource != null)
+            {
+                cancellationTokenSource.Cancel();
+                cancellationTokenSource.Dispose();
+                cancellationTokenSource = null;
+            }
+
+            var t = runningTentacleTask;
+            runningTentacleTask = null;
+
+            var stopDuration = Stopwatch.StartNew();
+            while (t?.IsCompleted == false && stopDuration.Elapsed < TimeSpan.FromSeconds(10))
```

Comment:
@nathanwoctopusdeploy could this be:
```
await When.Any(t, Task.Delay(10);
```
with a try catch to ignore any errors?

Reply:
no as it's not async and I wanted to avoid sync to async in the dispose